sdk/core/azure-core/tests/test_retry_policy.py

```diff
@@ -159,3 +160,31 @@ def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineRe
         pipeline = Pipeline(MockTransport(), [http_retry])
         pipeline.run(http_request)
     os.unlink(f.name)
+
+def test_retry_max_timeout():
+    class MockTransport(HttpTransport):
+        def __init__(self):
+            self.count = 0
+
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            pass
+        def close(self):
+            pass
+        def open(self):
+            pass
+
+        def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineResponse
+            self.count += 1
+            if self.count > 2:
+                assert self.count <= 2
+            time.sleep(5)
+            raise ServiceResponseError('timeout')
+
+    http_request = HttpRequest('GET', 'http://127.0.0.1/')
+    headers = {'Content-Type': "multipart/form-data"}
+    http_request.headers = headers
+    http_retry = RetryPolicy(retry_total=1, max_timeout=10)
+    pipeline = Pipeline(MockTransport(), [http_retry])
+    with pytest.raises(ServiceResponseError):
+        pipeline.run(http_request)
```

Comment:
Wouldn't the test be safer if:

* `retry_total` was larger to make sure it doesn't trigger before the timeout condition
* the exception was checked for the specific text `"timeout"`